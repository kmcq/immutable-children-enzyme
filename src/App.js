import React, { Component } from 'react';
import { List } from 'immutable';

class App extends Component {
  render() {
    return (
      <div className="App">
        {List(['first', 'second', 'third']).map(item => <span key={item}>{item}</span>)}
      </div>
    );
  }
}

export default App;
