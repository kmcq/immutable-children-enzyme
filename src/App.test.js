import React from 'react';
import App from './App';
import { shallow as enzyme2shallow } from 'enzyme2';
import { shallow as enzyme3shallow } from 'enzyme';

it('renders immutable children in enzyme 2', () => {
  expect(enzyme2shallow(<App />).debug()).toMatchSnapshot();
});

it('does not render immutable children in enzyme 3', () => {
  expect(enzyme3shallow(<App />).debug()).toMatchSnapshot();
});
