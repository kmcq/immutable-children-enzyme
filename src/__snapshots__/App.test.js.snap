// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render immutable children in enzyme 3 1`] = `
"<div className=\\"App\\">
  <undefined />
</div>"
`;

exports[`renders immutable children in enzyme 2 1`] = `
"<div className=\\"App\\">
  <span>
    first
  </span>
  <span>
    second
  </span>
  <span>
    third
  </span>
</div>"
`;
